docs/paper/dev/api/event-api/chat-event.md

```diff
@@ -0,0 +1,152 @@
+---
+slug: /dev/chat-events
+description: An outline on the AsyncChatEvent and how to handle it 
+---
+
+# Chat Events
+The Chat Event has evolved a few times over the years. 
+This guide will explain how to properly use the new `AsyncChatEvent` and its `ChatRenderer`.
+The `AsyncChatEvent` is an improved version of the old `AsyncPlayerChatEvent` that allows you to render chat messages individually for each player.
+
+## Understanding the Renderer
+
+Before we can start using the new chat event, we need to understand how the new renderer works.
+The renderer is Paper's way of allowing plugins to modify the chat message before it is sent to the player.
+This is done by using the `ChatRenderer` interface with its `render` method. Previously, this was done by using the `AsyncPlayerChatEvent` with its `setFormat` method.
+
+```java title="ChatRenderer#render"
+public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+    // ...
+}
+```
+
+- The `render` method is called when a chat message is sent to the player.
+- The `source` parameter is the player that sent the message.
+- The `sourceDisplayName` parameter is the display name of the player that sent the message.
+- The `message` parameter is the message that was sent.
+- The `viewer` parameter is the player that is receiving the message.
+
+Now that we understand how the renderer works, we can start using it.
+
+## Using the Renderer
+
+There are two ways to use the renderer.
+1. Creating a new Class that implements `ChatRenderer`. 
+2. Using a lambda expression.
+
+Depending on the complexity of your renderer you may want to use one or the other.
+
+### Creating a new Class
+
+If you want to use a class for your renderer, you just need to implement the `ChatRenderer` interface.
+
+```java title="ExampleRenderer.java"
+public class ExampleRenderer implements ChatRenderer {
+
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        // ...
+    }
+
+}
+```
+
+To be able to use this renderer, we need to tell the event to use it.
+
+```java title="ChatListner.java"
+public class ChatListner implements Listener {
+
+    @EventHandler
+    public void onChat(AsyncChatEvent event) {
+        event.renderer(new ExampleRenderer());
```

Comment:
Surely it's worth mentioning that you only need to instantiate this once? Or have `ChatListener` implement `ChatRenderer` and just pass `this`? Idk why we'd recommend instantiating a new object every time for no reason

Reply:
good point